spicy/runtime/include/parser.h

```diff
@@ -76,6 +77,57 @@ inline std::string to_string(const spicy::rt::ParserPort& x, adl::tag /*unused*/
 
 namespace spicy::rt {
 
+namespace detail {
+
+// Helper traits to detect whether a parser implements sink functions.
+
```

Comment:
can we avoid the repitition here create and create a helper the hides the boilerplate common to all four?